openqa-label-known-issues

```diff
@@ -136,31 +138,33 @@ investigate_issue() {
     echo "$reason" >> "$out"
     if [[ "$curl_output" != "200" ]] && [[ "$curl_output" != "301" ]]; then
         # if we can not even access the page it is something more critical
-        handle_unreachable "$testurl" "$out"
-        # Assume autoinst-log.txt still not found but process job anyway due to reason var
-        if [[ -n $reason ]] && [[ $curl_output = "404" ]] && [[ $reason != null ]]; then
-            label_on_issues_from_issue_tracker "$id"
+        handle_unreachable "$testurl" "$out" || return
+        # If we something other than a 404, we should not continue
+        [[ $curl_output != 404 ]] && return
         # Checking timestamp
-        elif [[ $(date -uIs -d '-14days') > $(grep timeago "$out" | hxselect -s '\n' -c '.timeago::attr(title)') ]]; then
+        if [[ $(date -uIs -d '-14days') > $(grep timeago "$out" | hxselect -s '\n' -c '.timeago::attr(title)') ]]; then
             # if the page is there but not even an autoinst-log.txt exists
             # then the job might be too old and logs are already deleted.
-            echoerr "'$testurl' does not have autoinst-log.txt but is rather old, ignoring"
+            echoerr "'$testurl' seems like an old job without autoinst-log.txt"
+        fi
+        # not unreachable, no log, no reason, not too old
+        if [[ -z $reason ]] || [[ $reason = null ]]; then
+            echoerr "'$testurl' does not have autoinst-log.txt or reason, cannot label"
             return
         fi
-    elif label_on_issues_from_issue_tracker "$id"; then
-        return
+    fi
+
+    label_on_issues_from_issue_tracker "$id" && return
 
     ## Issues without tickets, e.g. potential singular, manual debug jobs,
     # wrong user settings, etc.
     # could create an issue automatically with
     # $client_prefix curl -s -H "Content-Type: application/json" -X POST -H "X-Redmine-API-Key: $(sed -n 's/redmine-token = //p' ~/.query_redminerc)" --data '{"issue": {"project_id": 36, "category_id": 152, priority_id: 5, "subject": "test from command line"}}' https://progress.opensuse.org/issues.json
     # but we should check if the issue already exists, e.g. same
     # subject line
-    elif label_on_issues_without_tickets "$id"; then
-        return
-    else
-        handle_unreviewed "$testurl" "$out" "$reason" "$group_id" "$email_unreviewed" "$from_email" "$notification_address" "$job_data" "$dry_run"
-    fi
+    label_on_issues_without_tickets "$id" && return
+
```

Comment:
```suggestion
```

Reply:
I disagree.
We have a couple of top level `if`s here and I think it makes sense to add a blank line between all of them. And the final `handle_unreviewed`.
We were all struggling to understand the logic of the code.